python36Packages.dlib: 19.13 -> 19.16

r-ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools. This update was made based on information from https://repology.org/metapackage/python3.6-dlib/versions.



meta.description for python36Packages.dlib is: '"A general purpose cross-platform C++ machine learning library"'.

[Release on GitHub](https://github.com/davisking/dlib/releases/tag/v19.16)


[Compare changes on GitHub](https://github.com/davisking/dlib/compare/v19.13...v19.16)

<details>
<summary>
Checks done (click to expand)
</summary>

- built on NixOS

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 19.16 with grep in /nix/store/dbiymvay53xnz4vhipwz44dz25ylyxh5-python3.6-dlib-19.16
- directory tree listing: https://gist.github.com/3a286150f991eeb64fc59480a215a7c6
- du listing: https://gist.github.com/fbe21d22bf51199dfd54d0a3dc7b9a62

</details>
<details>
<summary>
Outpath report (click to expand)
</summary>

Outpath difference report
17 total rebuild paths

5 package rebuilds

5 x86_64-linux rebuilds
5 i686-linux rebuilds
2 x86_64-darwin rebuilds
5 aarch64-linux rebuilds


First ten rebuilds by attrpath
dlib
python27Packages.dlib
python27Packages.face_recognition
python36Packages.dlib
python36Packages.face_recognition

</details>

<details>
<summary>
Instructions to test this update (click to expand)
</summary>

Either download from Cachix:
```
nix-store -r /nix/store/dbiymvay53xnz4vhipwz44dz25ylyxh5-python3.6-dlib-19.16 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)

Or, build yourself:
```
nix-build -A python36Packages.dlib https://github.com/r-ryantm/nixpkgs/archive/c41603ab1b4e3e6c4314be22080355234d6aa0f6.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/dbiymvay53xnz4vhipwz44dz25ylyxh5-python3.6-dlib-19.16
ls -la /nix/store/dbiymvay53xnz4vhipwz44dz25ylyxh5-python3.6-dlib-19.16/bin
```


</details>
<br/>


cc @christopherpoole @ma27 for testing.

r-ryantm committed:
python36Packages.dlib: 19.13 -> 19.16

Semi-automatic update generated by
https://github.com/ryantm/nixpkgs-update tools. This update was made
based on information from
https://repology.org/metapackage/python3.6-dlib/versions

GrahamcOfBorg commented:
<!--REQUEST_ID=f43e3aa3-3fb1-4fc4-a934-b9b17b9eef11-->
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49070&attempt_id=c8091ae7-c365-4a89-8f9d-22bbf4812279)

The following builds were skipped because they don't evaluate on x86_64-darwin: python36Packages.dlib

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=f43e3aa3-3fb1-4fc4-a934-b9b17b9eef11-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49070&attempt_id=80ea3166-7dc1-4027-8a8f-3b02438de586)

Attempted: python36Packages.dlib

<details><summary>Partial log (click to expand)</summary><p>

```
tools/python/test/test_point.py ...                                      [ 43%]
tools/python/test/test_range.py .....                                    [ 50%]
tools/python/test/test_rgb_pixel.py .                                    [ 52%]
tools/python/test/test_sparse_vector.py ....                             [ 57%]
tools/python/test/test_svm_c_trainer.py ............                     [ 74%]
tools/python/test/test_vector.py ..................                      [100%]

===================== 66 passed, 5 skipped in 1.76 seconds =====================
pytestcachePhase
/nix/store/pj4pba1xn728bq2imyncl9jdsxfn985x-python3.6-dlib-19.16
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=f43e3aa3-3fb1-4fc4-a934-b9b17b9eef11-->
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49070&attempt_id=de9db4f1-6022-4611-a092-09e23cfff8ef)

Attempted: python36Packages.dlib

<details><summary>Partial log (click to expand)</summary><p>

```
[ 97%] Building CXX object CMakeFiles/dlib_python.dir/src/numpy_returns.cpp.o
[ 98%] Building CXX object CMakeFiles/dlib_python.dir/src/line.cpp.o
[100%] Linking CXX shared module /build/source/dlib.cpython-36m-x86_64-linux-gnu.so
[100%] Built target dlib_python
============================= test session starts ==============================
platform linux -- Python 3.6.6, pytest-3.7.4, py-1.5.4, pluggy-0.7.1
rootdir: /build/source, inifile:
/nix/store/9pifap75pkb9ifmdqh0irn8wgc3q38s6-stdenv-linux/setup: line 1298:  2835 Illegal instruction     /nix/store/bwfygfcdvis9wd1c1v51xwnwhw1hx0a0-python3-3.6.6/bin/python3.6m nix_run_setup test
builder for '/nix/store/c8idrynyfdfgnj357zjjs5kq2d0hsqj2-python3.6-dlib-19.16.drv' failed with exit code 132
error: build of '/nix/store/c8idrynyfdfgnj357zjjs5kq2d0hsqj2-python3.6-dlib-19.16.drv' failed
```
</p></details>

Ma27 commented:
not sure if I have sufficient time to test this tonight, but I'd like to have a deeper look at this (IIRC last time `dlib` caused some issues after an upgrade)

Ma27 commented:
the entire setup is somehow weird: although we have a working artifact on Hydra I can break the build of `dlib-19.13-python3.6` by appending a noop in a hook to alter the derivation hash.

Building this manually using `nix-shell -A` works perfectly fine though.

Ma27 commented:
adding `--no USE_AVX_INSTRUCTIONS` fixes the issue btw. Will push the change to @r-ryantm's branch later to see if borg succeeds then as well

Ma27 committed:
pythonPackages.dlib: disable AVX instructions during checkPhase

GrahamcOfBorg commented:
<!--REQUEST_ID=e84598db-3900-4e4a-a1d5-32cec8a6ee75-->
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49070&attempt_id=af89f21a-4a5a-4555-ac5c-f0f1e67860d6)

The following builds were skipped because they don't evaluate on x86_64-darwin: python36Packages.dlib

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=e84598db-3900-4e4a-a1d5-32cec8a6ee75-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49070&attempt_id=5005df25-3801-4bfe-b178-7017a4737347)

Attempted: python36Packages.dlib

<details><summary>Partial log (click to expand)</summary><p>

```
tools/python/test/test_point.py ...                                      [ 43%]
tools/python/test/test_range.py .....                                    [ 50%]
tools/python/test/test_rgb_pixel.py .                                    [ 52%]
tools/python/test/test_sparse_vector.py ....                             [ 57%]
tools/python/test/test_svm_c_trainer.py ............                     [ 74%]
tools/python/test/test_vector.py ..................                      [100%]

===================== 66 passed, 5 skipped in 1.99 seconds =====================
pytestcachePhase
/nix/store/0vkm30ys10rvjsn4ax3ar2x176pbi094-python3.6-dlib-19.16
```
</p></details>

Ma27 committed:
pythonPackages.dlib: move expression

Initially the expression was quite small (just inherited properties from
`pkgs.dlib`), but the more it grows the better it is to store it into
its own file.

Ma27 commented:
The change looks fine now and should be safe to merge.
I'll wait about 24h now in case anyone wants to review and request changes, then I'd merge this to master.

GrahamcOfBorg commented:
<!--REQUEST_ID=b588a029-eab0-4897-965f-793f8f1c7499-->
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49070&attempt_id=25baedcc-ec1e-431c-8083-de975a4ed68d)

The following builds were skipped because they don't evaluate on x86_64-darwin: python36Packages.dlib

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=b588a029-eab0-4897-965f-793f8f1c7499-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49070&attempt_id=719b9494-b7aa-4df8-a692-d3e4f08e07b9)

Attempted: python36Packages.dlib

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/0vkm30ys10rvjsn4ax3ar2x176pbi094-python3.6-dlib-19.16
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=b588a029-eab0-4897-965f-793f8f1c7499-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49070&attempt_id=25f26ed0-006a-41e4-ae06-617b6458c37e)

Attempted: python36Packages.dlib

<details><summary>Partial log (click to expand)</summary><p>

```
tools/python/test/test_point.py ...                                      [ 43%]
tools/python/test/test_range.py .....                                    [ 50%]
tools/python/test/test_rgb_pixel.py .                                    [ 52%]
tools/python/test/test_sparse_vector.py ....                             [ 57%]
tools/python/test/test_svm_c_trainer.py ............                     [ 74%]
tools/python/test/test_vector.py ..................                      [100%]

===================== 66 passed, 5 skipped in 0.92 seconds =====================
pytestcachePhase
/nix/store/7xing55bz8kzfr0ar9rv7r1gm2sri8mh-python3.6-dlib-19.16
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=e84598db-3900-4e4a-a1d5-32cec8a6ee75-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49070&attempt_id=1eb02649-76d0-4893-8adf-c25106bab8bb)

Attempted: python36Packages.dlib

<details><summary>Partial log (click to expand)</summary><p>

```
tools/python/test/test_point.py ...                                      [ 43%]
tools/python/test/test_range.py .....                                    [ 50%]
tools/python/test/test_rgb_pixel.py .                                    [ 52%]
tools/python/test/test_sparse_vector.py ....                             [ 57%]
tools/python/test/test_svm_c_trainer.py ............                     [ 74%]
tools/python/test/test_vector.py ..................                      [100%]

===================== 66 passed, 5 skipped in 3.21 seconds =====================
pytestcachePhase
/nix/store/7xing55bz8kzfr0ar9rv7r1gm2sri8mh-python3.6-dlib-19.16
```
</p></details>
